### ISSUE TITLE
Problem with 'aws s3 put-object' on newly created buckets

### ISSUE THREAD

**Issue by simon-wenmouth** [aws/aws-cli#223]
### TL;DR;

I've created a [gist/script to reproduce](https://gist.github.com/simon-wenmouth/6266773) the issue.

This should succeed (default region):

> ./aws-cli-issue-223.sh

This should loop forever

> ./aws-cli-issue-223.sh -r us-west-2

The script has command line options for AWS profile (-p), no ssl verification (-n), debug (-d), bash verbosity (-v) and AWS region (-r).
### Description

I am experiencing a large window of time (minutes to hours hours) between creating a bucket and being able to put an object in that bucket with the AWS CLI.  The problem seems to be related to specifying a location constraint for the bucket.

Here is a sequence of commands and outputs that reproduce the issue.  If the JSON output is long I'll truncate it with "...  snip ..."

First details on my system.

```
> bash --version
GNU bash, version 4.2.25(1)-release (x86_64-pc-linux-gnu)

> lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 12.04.2 LTS
Release:    12.04
Codename:   precise

> uname -a
Linux swenmouth04 3.5.0-37-generic #58~precise1-Ubuntu SMP Wed Jul 10 17:48:11 UTC 2013 x86_64 x86_64 x86_64 GNU/Linux

> aws --version
aws-cli/0.14.1 Python/2.7.3 Linux/3.5.0-37-generic
```

I start by creating a new bucket.

```
> cat "wobble" > wibble.txt
> aws s3 create-bucket --bucket $(openssl rand -hex 8) --create-bucket-configuration '{"LocationConstraint": "us-west-2"}'

> aws s3 list-buckets
{
    "Owner": {
        "DisplayName": "simon", 
        "ID": "dcdaddd0784ce3cbaa95023eb8837eb5a8a50f73bfb946f6d5dc550e14c22d12"
    }, 
    "Buckets": [
        {
            "CreationDate": "2013-08-19T06:43:44.000Z", 
            "Name": "40800480f3e11d30"
        }, 
        ... snip ...
    ]
}
```

I now attempt to put a file into the bucket.

```
> aws s3 put-object --bucket 40800480f3e11d30 --key wibble.txt --body wibble.txt --content-type text/plain --no-verify-ssl
{
    "Errors": [
        {
            "HostId": "HngxW88oiAlZDndfQYbXlQHPkudQnuRkATXg5n585iQMi3uwTSpdwFJwCdFWFBo6", 
            "Message": "Your socket connection to the server was not read from or written to within the timeout period. Idle connections will be closed.", 
            "Code": "RequestTimeout", 
            "RequestId": "95578A5CECA87D58"
        }
    ], 
    "ResponseMetadata": {}
}
A client error (RequestTimeout) occurred: Your socket connection to the server was not read from or written to within the timeout period. Idle connections will be closed.
```

I can however, put the file into the bucket with `s3cmd`.

```
> s3cmd put wibble.txt s3://40800480f3e11d30
wibble.txt -> s3://40800480f3e11d30/wibble.txt  [1 of 1]
 7 of 7   100% in    0s    39.45 B/s  done
WARNING: Redirected to: 40800480f3e11d30.s3-us-west-2.amazonaws.com
wibble.txt -> s3://40800480f3e11d30/wibble.txt  [1 of 1]
 7 of 7   100% in    0s    14.86 B/s  done
> aws s3 list-objects --bucket 40800480f3e11d30
{
    "CommonPrefixes": [], 
    "Contents": [
        {
            "LastModified": "2013-08-19T06:48:07.000Z", 
            "ETag": "\"3e6cb75a29d570bef41cf7ce9a11bcb6\"", 
            "StorageClass": "STANDARD", 
            "Key": "wibble.txt", 
            "Owner": {
                "DisplayName": "simon", 
                "ID": "dcdaddd0784ce3cbaa95023eb8837eb5a8a50f73bfb946f6d5dc550e14c22d12"
            }, 
            "Size": 7
        }
    ]
}
```

I can get the object and delete it too.

```
> aws s3 get-object --bucket 40800480f3e11d30 --key wibble.txt wibble-get.txt
{
    "AcceptRanges": "bytes", 
    "ContentType": "text/plain", 
    "LastModified": "Mon, 19 Aug 2013 06:48:07 GMT", 
    "ContentLength": "7", 
    "ETag": "\"3e6cb75a29d570bef41cf7ce9a11bcb6\""
}
> diff wibble.txt wibble-get.txt 
> aws s3 delete-object --bucket 40800480f3e11d30 --key wibble.txt

> aws s3 list-objects --bucket 40800480f3e11d30
{
    "CommonPrefixes": [], 
    "Contents": []
}
```

The problem is with, at least until the situation naturally resolves itself in however many hours/days (i'm not sure), the put-object command.  The buckets into which the put-object command works were created at least a day before.

If I remove the location constraint from the create-bucket command (so it is created in the US Standard zone) the put-object command is immediately successful.

Looking at [Buckets and Regions](http://docs.aws.amazon.com/AmazonS3/latest/dev/LocationSelection.html) in the AWS documentation it says that US-WEST-2 has read-after-write consistency (unlike US Standard) which means that new writes should be available immediately to all clients.  In, short, I would expect to have to wait between commands in US Standard and not in US West Oregon.

I'm not sure if this is an issue with the AWS CLI (since S3CMD works immediately on the bucket with the location constraint) or with the underlying S3 service the CLI depends upon.
